### PR TITLE
2755101 Improve access callback logic

### PR DESCRIPTION
**Pull request by haringsrob**

### CODE REVIEW

File: modules/checkout/tests/src/Functional/CheckoutOrderTest.php [hunk 1]
```diff
@@ -66,6 +71,66 @@ protected function setUp() {
...
+      'line_items' => [$line_item],
+    ]);
+    $order->save();
+
+    // I should not have access as anonymous.
+    $this->drupalLogout();
+    $this->drupalGet('/checkout/' . $order->id());
+    $this->assertSession()->statusCodeEquals(403);
+
+    // I should have access as the user that owns the order.
```
Comment (mglaman) [inline modules/checkout/tests/src/Functional/CheckoutOrderTest.php:116]: Should we test other users can't access the order, not just anon + owner?

Comment (haringsrob) [inline modules/checkout/tests/src/Functional/CheckoutOrderTest.php:116]: Yea why not, it covers a potential security issue. I'll add it

Comment (haringsrob) [inline modules/checkout/tests/src/Functional/CheckoutOrderTest.php:116]: I have updated the test.

**Comment by bojanz** [comment]
Committed in 857f69da3f3bc89d443d1b5c8969249f4cd68dc5.

**Pull request closed without merging**